src/code.c

```diff
@@ -95,29 +97,20 @@ static inline void PopLoopNesting( void ) {
   STATE(LoopNesting) = STATE(LoopStack)[--STATE(LoopStackCount)];
 }
 
-static inline void SetupGapname(TypInputFile* i)
+static inline UInt SetupGapname(void)
 {
-    if (i->gapnameid == 0) {
-        Obj filename = MakeImmString(i->name);
-#ifdef HPCGAP
-        // TODO/FIXME: adjust this code to work more like the corresponding
-        // code below for GAP?!?
-        i->gapnameid = AddAList(FilenameCache, filename);
-#else
-        Obj pos = POS_LIST(FilenameCache, filename, INTOBJ_INT(1));
-        if (pos == Fail) {
-            i->gapnameid = PushPlist(FilenameCache, filename);
-        }
-        else {
-            i->gapnameid = INT_INTOBJ(pos);
-        }
-#endif
-    }
+    const Char * name = GetInputFilename();
+    UInt         rnam = RNamName(name);
+    if (!ISB_REC(FilenameCache, rnam))
+        ASS_REC(FilenameCache, rnam, True);
```

Comment:
@ChrisJefferson This change to `SetupGapname` might concern you most: I replaced the `gapnameid` field in `TypInputFile` by the use of RNams, which avoids most HPC-GAP issues. The drawback is a slight overhead for each new function definition we parse. So this would mostly impact startup times, but I found no measurable regression there (given that we only do this a couple 10,000 times, I didn't expect any).

The code in `profile.c` was not changed to adapt for this in any way; which means that `OutputtedFilenameList` now will be larger than it needs to be (e.g. for me, there are 642 entries in `GET_FILENAME_CACHE()`, but their rnams range up to 10025 -- so we waste about 80kb of RAM, which I found acceptable. Of course we could avoid this; e.g. we could use an atomic counter to assign an ascending value to each filename. We'd then need two objects to store the "cache": One would be a precord, like the above, which maps the rnams to their unique ids; the second is a plist, which stores for each unique id the corresponding name as a string obj (i.e. store the result of `NAME_OBJ_RNAM`). Then, we use the unique id as file id. Voilà, sequential fileids.

But this seemed like an awful lot of effort to save a (by today's standards) measly 80 kb... Thoughts?

Reply:
@ChrisJefferson ping?

Reply:
One problem with this is that RNamNames are limited to 1023 characters (I've hit this while abusing records as string stores before), is that a reasonable limit to put on filenames?

Reply:
`GAP_PATH_MAX` is 4096...

Reply:
@ChrisJefferson That's of course an excellent point, I completely forgot about that :-(. 
Gotta think about alternatives then. In order to reduce the overall bitrot, I'd like to do the following: I'll drop the last three commits in this PR for now; then we can get all that merged ASAP, I hope; and I'll open a new PR with the remaining changes, which we can then further discuss (and fix, and what not all)